### DISCUSSION TITLE
Rebuild using Hakyll

### DISCUSSION THREAD

**Discussion by aerphanas** [learnyouahaskell/learnyouahaskell.github.io#42] [Ideas]
Hakyll is a static site generator using haskell, I don't know if I can do this or not, I think the code will be more maintainable and cleaner than the current code, what does the community think?

**Comment by MatthijsBlom** [comment]
Which code exactly do you expect this to replace?

**Reply by aerphanas** [reply]
idk, i just think it is good that this site generated using hakyll or maybe elm (?)

**Comment by pierluc-codes** [comment]
We are now with pandoc :)

**Reply by ulysses4ever** [reply]
Well, iit's more like: pandoc+a bunch of horrible sed+some shell. I wanted to port to Hakyll actually but I wanted to discuss it with you first.

**Reply by ulysses4ever** [reply]
And FTR, Hakyll uses pandoc under the hood.

**Reply by pierluc-codes** [reply]
I trust your judgment. If you feel it will be easier to maintain, you have my full support.

That being said, I would like to mention that we received some feedback that it would be cool to not only be limited to a website, [but have also a epub](https://github.com/learnyouahaskell/learnyouahaskell.github.io/issues/2).

**Reply by ulysses4ever** [reply]
Thank you, I appreciate your trust.

Epub will have to be looked into separately in any case I feel like. The power of Hakyll is that it's a simple convenience wrapper around Pandoc for the most part (to automate mundane tasks like copying, changing filename extensions, providing metadata to Pandoc, etc.). So, since Pandoc [supports epub](https://pandoc.org/epub.html), Hakyll can too. [Here's a thread](https://groups.google.com/g/hakyll/c/CW0WAfqUVwo?pli=1) discussing turning separate Hakyll posts (so, plain Markdown processed in the Hakyll framework) into EPUB, for instance.
